### PR TITLE
Replace @babel/polyfill with core-js and regenerator-runtime (#11743)

### PR DESCRIPTION
**Pull request by alexandec**
## Description
`@babel/polyfill` has been deprecated as of 7.4.0. This PR does the following:

- Replace @babel/polyfill with `core-js/stable` and `regenerator-runtime/runtime` for apps, unit tests, and E2E tests
- Upgrade core-js from version 2 to version 3 (latest)
- Regenerate the list of polyfills in `src/platform/polyfills/preESModulesPolyfills.js` (used by older browsers, had to be re-created due to the core-js upgrade)

## Testing done
Local smoke test, unit tests, E2E tests.

## Acceptance criteria
- [ ] App builds without errors and functions correctly
- [ ] All tests pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/polyfills/preESModulesPolyfills.js [hunk 1]
```diff
@@ -22,88 +22,140 @@ import 'classlist-polyfill'; // DOM element classList support.
  * 6. Now you have the full list and you can format them into import statements with some
  *   regex work
  */
```
Comment (alexandec) [inline src/platform/polyfills/preESModulesPolyfills.js:24]: I regenerated this list by following the instructions in the comment above

**Review by jhonny218 — APPROVED** [review]
LGTM

**Comment by alexandec** [comment]
Thanks @jhonnyoddball ! I'll request one more review just to be sure and then I think this is good.

File: src/platform/testing/unit/mocha.opts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 --require src/platform/testing/unit/mocha-setup.js
 --require src/platform/testing/unit/enzyme-setup.js
 --require choma
---require @babel/polyfill
+--require core-js/stable
+--require regenerator-runtime/runtime
```
Comment (U-DON) [inline src/platform/testing/unit/mocha.opts:5]: I think we're safe to remove this file since we're on Mocha 8 now, but that's out of scope here.

**Review by U-DON — APPROVED** [review]

**Review by billfienberg — APPROVED** [review]

File: src/platform/testing/unit/mocha.opts (hunk 2 above)
Comment (alexandec) [reply src/platform/testing/unit/mocha.opts:5]: Okay cool, I will keep the file in this PR but we can look at removing it in the future.

**Pull request merged**